### DISCUSSION TITLE
Coverage in combination with changed option.

### DISCUSSION THREAD

**Discussion by OssamaSijbesma** [vitest-dev/vitest#4868] [Q&A] [answered]
To improve our CI pipeline executing time, I would like to use the [changed](https://vitest.dev/guide/cli.html#changed) option which executes a git diff and only runs tests on the affected files. However, it seems that the coverage reports doesn't seem to make any sense: files are included that are not tested and the coverage % is below the thresholds but succeeds anyway. I was wondering if others managed to get it to work together.

Command:
`vitest run --coverage --silent --changed`

Coverage options:
```
coverage: {
          provider: 'istanbul',
          reportsDirectory: './coverage',
          cleanOnRerun: true,
          all: false,
          statements: 70,
          branches: 65,
          functions: 60,
          lines: 70,
          include: ['src/**/*.{vue,ts,js}'],
          exclude: [],
          reporter: ['cobertura', 'text', 'html'],
          extension: ['.js', '.ts', '.vue'],
        },
```

**Comment by AriPerkkio** [comment, marked as answer]
Currently `--changed` doesn't affect coverage in any way. If you have suggestions how it should behave, feel free to open feature request for this. There was also some similar discussion earlier here: https://github.com/vitest-dev/vitest/discussions/4057.

In your case you might get everything working nicely by using `--coverage.enabled --coverage.all=false` with the `--changed`. Then coverage report will only include the files that were imported by the test cases.

**Reply by OssamaSijbesma** [reply]
Thank you @AriPerkkio for clarifying, I might create a feature request somewhere in the future 🙂

**Comment by jackonwei** [comment]
Hey, I have a question, I want to know how you do it, after the coverage report is exported, only a single test of one file is written, but the other files are 100% by default, and the overall project project coverage does not include the untested files, only the coverage of the written single test file is calculated, which is not what I want
